### PR TITLE
79667 priceupdate event slicing implementation

### PR DESCRIPTION
**Pull request by mayur1501364**
1. Added slicing method to slice and generate event in webjob
2. Unit test project and added test cases for webjob
3. Created new Azure tables 
4. Uploaded sliced events to blob

### CODE REVIEW

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/AzureBlobStorageHelper.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
+﻿using System.ComponentModel;
+using System.IO.Abstractions;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
```
Comment (Shirin14926) [inline tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/AzureBlobStorageHelper.cs:5]: Remove unused namespaces

File: src/UKHO.ERPFacade.Common/Logging/EventIds.cs [hunk 2]
```diff
@@ -200,6 +200,73 @@ public enum EventIds
...
+
+        /// <summary>
+        /// 940053 - Bulk price information payload received from SAP
+        /// </summary>
+        SapBulkPriceInformationPayloadReceived = 940053,
+
+        /// <summary>
+        /// 940054 - Storing the received Bulk price information event in azure table.
+        /// </summary>
+        StoreBulkPriceInformationEventInAzureTable = 940054,
```
Comment (Shirin14926) [inline src/UKHO.ERPFacade.Common/Logging/EventIds.cs:270]: Comma can be removed

File: src/UKHO.ERPFacade.Common/Models/PriceChange.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace UKHO.ERPFacade.Common.Models
+{
```
Comment (Shirin14926) [inline src/UKHO.ERPFacade.Common/Models/PriceChange.cs:9]: Remove unused namespaces and kindly exclude this code from code coverage

Add [ExcludeFromCodeCoverage]

File: tests/UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/BulkPriceUpdateScenario.cs [hunk 4]
```diff
@@ -55,5 +55,17 @@ public async Task WhenValidEventReceivedWithInvalidToken_ThenBPUpdateReturns403U
 
         }
 
+        [Category("DevEnvFT")]
+        [Test, Order(1)]
+        [TestCase("PC1_Payload.JSON", TestName = "WhenValidEventReceivedWithValidToken_ThenBPUpdateReturn200OkResponse")]
```
Comment (Shirin14926) [inline tests/UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/BulkPriceUpdateScenario.cs:60]: Can be renamed as WhenValidBulkPriceEventReceivedWithValidToken_ThenBPUpdateReturns200OkResponse

File: tests/UKHO.ERPFacade.PublishPriceChange.WebJob.UnitTests/Services/SlicingPublishingServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+            && call.GetArgument<LogLevel>(0) == LogLevel.Information
+            && call.GetArgument<EventId>(1) == EventIds.UploadedSlicedEventInAzureBlob.ToEventId()
+            && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "Sliced event is uploaded in blob storage successfully.").MustHaveHappened();
+
+        }
+
+        [Test]
+        public void WhenUnitPriceEntitiesAreIncompleted_ThenShouldSliceAndPublish()
+        {
+            IList<PriceChangeMasterEntity> priceChangeMasterEntities = new List<PriceChangeMasterEntity>();
```
Comment (Shirin14926) [inline tests/UKHO.ERPFacade.PublishPriceChange.WebJob.UnitTests/Services/SlicingPublishingServiceTests.cs:119]: Can this be refactored as,
IList<PriceChangeMasterEntity> priceChangeMasterEntities = new List<PriceChangeMasterEntity>
            {
                new()
                {

File: tests/UKHO.ERPFacade.PublishPriceChange.WebJob.UnitTests/Services/SlicingPublishingServiceTests.cs [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+                CorrId = "FakeCorrID",
+                ETag = new(),
+                PartitionKey = "FakeCorrID",
+                RowKey = "FakeCorrID",
+                Status = "Incomplete",
+                Timestamp = new DateTimeOffset()
+            });
+            IList<UnitPriceChangeEntity> unitPriceChangeEntities = new List<UnitPriceChangeEntity>();
+            unitPriceChangeEntities.Add(new UnitPriceChangeEntity()
+            {
```
Comment (Shirin14926) [inline tests/UKHO.ERPFacade.PublishPriceChange.WebJob.UnitTests/Services/SlicingPublishingServiceTests.cs:131]: Similarly,
 IList<UnitPriceChangeEntity> unitPriceChangeEntities = new List<UnitPriceChangeEntity>
            {
                new UnitPriceChangeEntity()
                {

File: tests/UKHO.ERPFacade.PublishPriceChange.WebJob.UnitTests/Services/SlicingPublishingServiceTests.cs [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+                Eventid = "FakeEventID",
+                MasterCorrid = "FakeCorrID",
+                RowKey = "FakeEventID",
+                ETag = new(),
+                PartitionKey = "FakeEventID",
+                Status = "Incomplete",
+                Timestamp = new DateTimeOffset(),
+                UnitName = "FakeUnitName"
+            });
+            unitPriceChangeEntities.Add(new UnitPriceChangeEntity()
```
Comment (Shirin14926) [inline tests/UKHO.ERPFacade.PublishPriceChange.WebJob.UnitTests/Services/SlicingPublishingServiceTests.cs:141]: can be changed to - new UnitPriceChangeEntity()

**Review by Shirin14926 — COMMENTED** [review]
Few minor comments please

File: src/UKHO.ERPFacade.Common/Logging/EventIds.cs (hunk 2 above)
Comment (mayur1501364) [reply src/UKHO.ERPFacade.Common/Logging/EventIds.cs]: removed comma

File: src/UKHO.ERPFacade.Common/Models/PriceChange.cs (hunk 3 above)
Comment (mayur1501364) [reply src/UKHO.ERPFacade.Common/Models/PriceChange.cs:5]: Added [ExcludeFromCodeCoverage]

File: tests/UKHO.ERPFacade.PublishPriceChange.WebJob.UnitTests/Services/SlicingPublishingServiceTests.cs (hunk 5 above)
Comment (mayur1501364) [reply tests/UKHO.ERPFacade.PublishPriceChange.WebJob.UnitTests/Services/SlicingPublishingServiceTests.cs]: Simplified

File: tests/UKHO.ERPFacade.PublishPriceChange.WebJob.UnitTests/Services/SlicingPublishingServiceTests.cs (hunk 7 above)
Comment (mayur1501364) [reply tests/UKHO.ERPFacade.PublishPriceChange.WebJob.UnitTests/Services/SlicingPublishingServiceTests.cs]: changed to new UnitPriceChangeEntity()

**Review by Shirin14926 — APPROVED** [review]
Thanks @mayur1501364  & @Sadha1501493 for changes. LGTM

**Review by jrippington — APPROVED** [review]
Looks good

**Review by DanOwenUKHO — APPROVED** [review]

**Review by JiviteshT — APPROVED** [review]
Approved to merge.

**Pull request merged**